openhands/tools/browser_use/definition.py

```diff
@@ -34,9 +31,14 @@ def agent_observation(self) -> Sequence[TextContent | ImageContent]:
         if self.error:
             return [TextContent(text=f"Error: {self.error}")]
 
-        content: list[TextContent | ImageContent] = [
-            TextContent(text=maybe_truncate(self.output, MAX_BROWSER_OUTPUT_SIZE))
-        ]
+        # Use enhanced truncation with file saving, fallback to current working dir
+        truncated_text = maybe_truncate(
+            content=self.output,
+            truncate_after=DEFAULT_TEXT_CONTENT_LIMIT,
+            save_dir=os.getcwd(),
```

Comment:
We should accept a `save_dir` from browser tool set .create() instead of cwd so it doesn't get into VCS